csrc/executor_utils.cpp

```diff
@@ -1072,7 +1072,7 @@ class NvrtcCompileDriver {
       std::cout << log.data() << std::endl;
     }
 
-    return log;
+    return std::string(log.data());
```

Comment:
Why is this necessary?

Reply:
`return log` only returns a empty string. seems like a std::string issue, didn't dig deep.

Reply:
I'd be really surprised `std::string` has a bug.

Reply:
Please just don't assume the C++ standard library has a bug. Are you really sure we using the memory correctly?

Reply:
But isn't it filled by this?

```
NVRTC_SAFE_CALL(nvrtcGetProgramLog(program, log.data()));
```

Reply:
No. After `NVRTC_SAFE_CALL(nvrtcGetProgramLog(program, log.data()));` log itself is still empty.

Reply:
Why?

Reply:
`log.reserve(logsize);` just means a **planned** memory usage. The memory is reserved for it but it is **NOT used** yet. `log.data()` is just go to the reserved memroy space and write something. But nobody tells the string object the reserved memroy is now being used. The string object is still thinking I have some memory in the bank but I have nothing in my pocket.

Reply:
Oh, I thought `nvrtcGetProgramLog(program, log.data())` was enough since the buffer is null-terminated, but I didn't know that means nothing in C++ string.

This was helpful: https://akrzemi1.wordpress.com/2014/03/20/strings-length/

Can you also fix this? https://github.com/NVIDIA/Fuser/pull/359/files#diff-7a4f65bea86e65f941636f4355972029ce48141c38d5ae2db39c10ad7a291207R1107 It should be just replaced with `log_.resize(kLogSize)`

Reply:
done.